cmd/validators.go

```diff
@@ -51,9 +53,36 @@ func ValidateFromTo(src, dst string, userSpecifiedFromTo string) (common.FromTo,
 	return userFromTo, nil
 }
 
-const fromToHelpText = "Valid values are two-word phases of the form BlobLocal, LocalBlob etc.  Use the word 'Blob' for Blob Storage, " +
-	"'Local' for the local file system, 'File' for Azure Files, and 'BlobFS' for ADLS Gen2. " +
-	"If you need a combination that is not supported yet, please log an issue on the AzCopy GitHub issues list."
+const fromToHelpFormat = "Valid FromTo are pairs of Source-Destination words (e.g. BlobLocal, BlobBlob) that specify the source and destination resource types. All valid FromTos are: %s"
+
+var fromToHelp = func() string {
+	validFromTos := ""
+
+	isSafeToOutput := func(loc common.Location) bool {
+		switch loc {
+		case common.ELocation.Benchmark(),
+			common.ELocation.None(),
+			common.ELocation.Unknown():
+			return false
+		default:
+			return true
+		}
+	}
+
+	enum.GetSymbols(reflect.TypeOf(common.EFromTo), func(enumSymbolName string, enumSymbolValue interface{}) (stop bool) {
```

Comment:
Does this allow S3 and GCP to be printed as To?

Reply:
The full line reads as follows:

```
Valid FromTo are pairs of Source-Destination words (e.g. BlobLocal, BlobBlob) that specify the source and destination resource types. All valid FromTos are: BlobBlob, BlobFSLocal, BlobFile, BlobLocal, BlobPipe, FileBlob, FileFile, FileLocal, FilePipe, GCPBlob, LocalBlob, LocalBlobFS, LocalFile, PipeBlob, PipeFile, S3Blob
```

Reply:
It only iterates over the pre-specified FromTos; it does not create any from-tos itself.